Library/Homebrew/language/python.rb

```diff
@@ -94,21 +94,28 @@ def self.setup_install_args(prefix, python = "python3")
     module Shebang
       module_function
 
+      # A regex to match potential shebang permutations.
+      SHEBANG_REGEX = %r{^#! ?/usr/bin/(?:env )?python(?:[23](?:\.\d{1,2})?)?( |$)}.freeze
```

Comment:
If you're feeling generous, I'd love for `rewrite_shebang` to be able to handle [this](https://github.com/Homebrew/homebrew-core/blob/e3003ff878e70b58d466476a1e3fb21cf4b5fea5/Formula/nmap.rb#L69-L72):
```ruby
# We can't use `rewrite_shebang` here because `detected_python_shebang` only works
# for shebangs that start with `/usr/bin`, but the shebang we want to replace
# might start with `/Applications` (for the `python3` inside Xcode.app).
inreplace bin/"ndiff", %r{\A#!.*/python(\d+(\.\d+)?)?$}, "#!/usr/bin/env python3"
```